var/spack/repos/builtin/packages/synapsetool/package.py

```diff
@@ -66,12 +65,14 @@ def libs(self):
 
     def cmake_args(self):
         args = []
-        if self.spec.satisfies('+mpi'):
+        if self.spec.satisfies('^mpi'):
             args.extend([
                 '-DCMAKE_C_COMPILER:STRING={}'.format(self.spec['mpi'].mpicc),
                 '-DCMAKE_CXX_COMPILER:STRING={}'.format(self.spec['mpi'].mpicxx),
-                '-DSYNTOOL_WITH_MPI:BOOL=ON',
             ])
+        if self.spec.satisfies('+mpi'):
+            args.extend(['-DSYNTOOL_WITH_MPI:BOOL=ON'])
+
```

Comment:
not sure why above change was required.  the original implementation with `+mpi` was sufficient.

Reply:
In case we are compiling a non-mpi version but it depends on something with mpi.